### PR TITLE
Reduce deployer notifications

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] makes call to git more resilient with basic retrial
- [x] the deployer should not crash anymore when a git/docker are not responding but wait 5 minutes and try again.

## Related issue number
#12 

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/deployment-agent/src/simcore_service_deployment_agent/notifier.py [hunk 1]
```diff
@@ -30,9 +33,59 @@
...
+    if mattermost_config["enabled"]:
+        status_emoji = ":+1: "
+        if state is State.FAILED:
+            status_emoji = ":x:"
+        elif state is State.PAUSED:
+            status_emoji = ":x:"
+
+        header_unique_name = mattermost_config["header_unique_name"]
+        date = datetime.datetime.utcnow()
+        message = "{} {} {} - {} |".format(header_unique_name, status_emoji, status_message, date.isoformat(timespec='seconds'))
```
Comment (pcrespov) [inline services/deployment-agent/src/simcore_service_deployment_agent/notifier.py]: MINOR: py3 allows formatted strings as ``f"{header_unique_name}  {status_emoji} ... "`` makes code shorter :-)

File: services/deployment-agent/src/simcore_service_deployment_agent/notifier.py [hunk 2]
```diff
@@ -30,9 +33,59 @@
...
+            new_header = message
+            start_index = current_header.find(header_unique_name)
+            if start_index != -1:
+                # update the message instead
+                lastindex = current_header.find("|", start_index)
+                new_header = "{}{}{}".format(current_header[0:start_index], message, current_header[lastindex+1:])
+
+            url = URL(mattermost_config["url"]).with_path("api/v4/channels/{}/patch".format(channel_id))
+            async with client.put(url, headers=headers, data=json.dumps({"header": new_header})) as resp:
+                log.debug("request response received with code %s", resp.status)
```
Comment (pcrespov) [inline services/deployment-agent/src/simcore_service_deployment_agent/notifier.py]: MINOR: add in log an explanation of what you requested

File: services/deployment-agent/src/simcore_service_deployment_agent/notifier.py (hunk 1 above)
Comment (sanderegg) [reply services/deployment-agent/src/simcore_service_deployment_agent/notifier.py]: done

File: services/deployment-agent/src/simcore_service_deployment_agent/notifier.py (hunk 2 above)
Comment (sanderegg) [reply services/deployment-agent/src/simcore_service_deployment_agent/notifier.py]: done

**Pull request merged**